### ISSUE TITLE
MapValues in field arguments are incorrectly serialized in the Java codegen

### ISSUE THREAD

**Issue by martinbonnin** [apollographql/apollo-kotlin#3857]
Follow up from https://community.apollographql.com/t/error-converting-to-json-when-using-normalized-cache-in-apollo-kotlin/2711/4

```graphql
mutation CreateCat {
  createAnimal(
    input: {
      class: MAMMAL
      name: "cat"
    }
  ) {
    ...catFragment
  }
}
```

generates:

```kotlin
  public static List<CompiledSelection> root = Arrays.asList(
    new CompiledField.Builder("createAnimal", Animal.type).arguments(Arrays.asList(new CompiledArgument("input", new ImmutableMapBuilder().put("class", "MAMMAL").put("name", "cat"), false))).selections(createAnimal).build()
  );
```

which doesn't work because there's a `build()` missing

**Comment by martinbonnin** [comment]
Awesome PR by @anderssandven and @BoD  is there https://github.com/apollographql/apollo-kotlin/pull/3861

If anyone comes into this, a workaround is to use a variable for the input object and define it at runtime:

```graphql
# Replace
mutation UpdateReviewWithoutVariable {
  updateReview(id: "0", review: {
    stars: 5
  }) {
    id
    stars
  }
}

# With
mutation UpdateReviewWithoutVariable($reviewInput: ReviewInput) {
  updateReview(id: "0", review: $reviewInput){
    id
    stars
  }
}
```

**Comment by martinbonnin** [comment]
Fixed in `3.2.0`
